docs/connectors/sources/delta.md

```diff
@@ -15,91 +15,136 @@ tolerance](..#fault-tolerance).
-   `snapshot_and_follow` modes, it follows transaction log records **after** this version.
-
-   Note: at most one of `version` and `datetime` options can be specified.
-   When neither of the two options is specified, the latest committed version of the table
-   is used.
-
-* `datetime` - Optional timestamp for the snapshot in the ISO-8601/RFC-3339 format, e.g.,
-   "2024-12-09T16:09:53+00:00". When this option is set, the connector finds and opens the
-   version of the table as of the specified point in time (based on the server time recorded
-   in the transaction log, not the event time encoded in the data).  In `snapshot` and
-   `snapshot_and_follow` modes, it retrieves the snapshot of this version of the table.
-   In `follow` and `snapshot_and_follow` modes, it follows transaction log records
-   **after** this version.
-
-   Note: at most one of `version` and `datetime` options can be specified.
-   When neither of the two options is specified, the latest committed version of the table
-   is used.
+| Property                    | Type   | Default    | Description   |
+|-----------------------------|--------|------------|---------------|
+| `uri`*                      | string |            | Table URI, e.g., "s3://feldera-fraud-detection-data/demographics_train"|
+| `mode`*                     | enum   |            | Table read mode. Three options are available: <ul> <li>`snapshot` - read a snapshot of the table and stop.</li> <li>`follow` - follow the changelog of the table, only ingesting changes (new and deleted rows)</li> <li>`snapshot_and_follow` - Read a snapshot of the table before switching to the `follow` mode.  This mode implements the backfill pattern where we load historical data for the table before ingesting the stream of real-time updates.</li> </ul>|
+| `timestamp_column`          | string |            | Table column that serves as an event timestamp. When this option is specified, and `mode` is one of `snapshot` or `snapshot_and_follow`, table rows are ingested in the timestamp order, respecting the [`LATENESS`](/sql/streaming#lateness-expressions) property of the column: each ingested row has a timestamp no more than `LATENESS` time units earlier than the most recent timestamp of any previously ingested row.  See details [below](#ingesting-time-series-data-from-a-delta-lake). |
+| `snapshot_filter`           | string |            | <p>Optional row filter.  This option is only valid when `mode` is set to `snapshot` or `snapshot_and_follow`.  When specified, only rows that satisfy the filter condition are included in the snapshot.  The condition must be a valid SQL Boolean expression that can be used in the `where` clause of the `select * from snapshot where ..` query.</p><p> This option can be used to specify the range of event times to include in the snapshot, e.g.: `ts BETWEEN TIMESTAMP '2005-01-01 00:00:00' AND TIMESTAMP '2010-12-31 23:59:59'`.</p>
```

Comment:
I suspect the condition cannot contain calls to user-defined functions, for example.
Can it contain calls to other functions?

Reply:
yes, it can be anything that datafusion can compute. I'll explain it in the docs